### PR TITLE
[ROCm] add AMD devices info for MI210, MI100 and MI50/60

### PR DESCRIPTION
**Pull request by i-chaochen**
This PR brings back gpu_device_info_test  for ROCm and add AMD gpu device info on MI210, MI100 and MI50/60

Thanks

### CODE REVIEW

**Comment by i-chaochen** [comment]
Hi @akuegel wonder will you also be able to review this PR as well?

Thanks in advance!

File: xla/stream_executor/rocm/rocm_gpu_executor.h
Comment (akuegel) [inline xla/stream_executor/rocm/rocm_gpu_executor.h:1]: nit: 2015 -> 2023?

File: xla/service/gpu/gpu_device_info_test.cc
Comment (akuegel) [inline xla/service/gpu/gpu_device_info_test.cc:38]: nit: line break? I think this is more than 80 characters.

File: xla/stream_executor/rocm/rocm_gpu_executor.h
Comment (i-chaochen) [reply xla/stream_executor/rocm/rocm_gpu_executor.h]: Yes! done!

**Review by akuegel — APPROVED** [review]

File: xla/service/gpu/gpu_device_info_test.cc
Comment (ddunl) [reply xla/service/gpu/gpu_device_info_test.cc:94]: As I write this now I'm not so sure. Let me know. I'm only considering this because our linter takes issue with how the braces end up with this change, and I'd prefer to not guard this with a `#if ... #endif` if possible. If it is necessary, I'll patch in the necessary `// NOLINT` lines internally

Comment (i-chaochen) [reply xla/service/gpu/gpu_device_info_test.cc:94]: Hi @ddunl Thanks a lot for the feedback.

Yes, this ```#if``` is required here. As prior to ROCm 5.5, the device_name cannot be printed out and we have to wait ROCm5.5 is released to fix this test.

Comment (ddunl) [reply xla/service/gpu/gpu_device_info_test.cc:94]: Gotcha, I may not get the internal patch done by the end of the day, if not it'll be early tomorrow. Thanks for letting me know!!

Comment (akuegel) [reply xla/service/gpu/gpu_device_info_test.cc:94]: I have adjusted the patch internally to have a // NOLINT at the end of these lines. There was also a missing std:: in front of string. With these fixes, it should be ready to land, I will just need to wait for an approval for the adjusted patch.

Comment (i-chaochen) [reply xla/service/gpu/gpu_device_info_test.cc:94]: @akuegel Thanks a lot!

**Pull request closed without merging**
